src/enums/staffPaymentsOptions.ts

```diff
@@ -1,5 +1,5 @@
 export enum StaffPaymentOptions {
-    NONE = NaN,
+    NONE = -1,
```

Comment:
This is because you can't check equality with NaN. (You have to use isNaN() instead.)

Now we can indeed check `if (val === StaffPaymentOptions.NONE)...`.